dashboard/src/components/Catalog/Catalog.v2.tsx

```diff
@@ -0,0 +1,152 @@
+import CatalogItems from "./CatalogItems";
+
+interface ICatalogProps {
+  charts: IChartState;
+  repo: string;
+  filter: string;
+  fetchCharts: (namespace: string, repo: string) => void;
+  pushSearchFilter: (filter: string) => RouterAction;
+  namespace: string;
+  kubeappsNamespace: string;
+  getCSVs: (namespace: string) => void;
+  csvs: IClusterServiceVersion[];
+  featureFlags: IFeatureFlags;
+}
+
+function Catalog(props: ICatalogProps) {
+  const {
+    charts: {
+      isFetching,
+      selected: { error },
+      items: charts,
+    },
+    fetchCharts,
+    namespace,
+    pushSearchFilter,
+    getCSVs,
+    csvs,
+    repo,
+    filter: propsFilter,
+  } = props;
+  const [searchFilter, setSearchFilter] = useState(propsFilter);
+  const [typeFilter, setTypeFilter] = useState([] as string[]);
+  const [repoFilter, setRepoFilter] = useState([] as string[]);
+  const [operatorProviderFilter, setOperatorProviderFilter] = useState([] as string[]);
+
+  const allRepos = uniq(charts.map(c => c.attributes.repo.name));
+  const allProviders = uniq(csvs.map(c => c.spec.provider.name));
+
+  useEffect(() => {
+    fetchCharts(namespace, repo);
+    getCSVs(namespace);
+  }, [namespace, repo, fetchCharts, getCSVs]);
+
+  useEffect(() => {
+    setSearchFilter(propsFilter);
+  }, [propsFilter]);
+
+  const filteredCharts = charts
+    .filter(() => typeFilter.length === 0 || typeFilter.includes("Charts"))
+    .filter(() => operatorProviderFilter.length === 0)
+    .filter(c => new RegExp(escapeRegExp(searchFilter), "i").test(c.id))
+    .filter(c => repoFilter.length === 0 || repoFilter.includes(c.attributes.repo.name));
+  const filteredCSVs = csvs
+    .filter(() => typeFilter.length === 0 || typeFilter.includes("Operators"))
+    .filter(() => repoFilter.length === 0)
+    .filter(c => new RegExp(escapeRegExp(searchFilter), "i").test(c.metadata.name))
+    .filter(
+      c =>
+        operatorProviderFilter.length === 0 ||
+        operatorProviderFilter.includes(c.spec.provider.name),
+    );
+
+  return (
+    <section>
+      <PageHeader>
+        <div className="kubeapps-header">
+          <Row>
+            <h1>Catalog</h1>
+            <SearchFilter
+              key="searchFilter"
+              placeholder="search charts..."
+              onChange={setSearchFilter}
+              value={searchFilter}
+              onSubmit={pushSearchFilter}
+            />
+          </Row>
+        </div>
+      </PageHeader>
+      <LoadingWrapper loaded={!isFetching}>
+        {error && (
+          <Alert theme="danger">Found en error fetching the catalog: {error.message}</Alert>
+        )}
+        {charts.length === 0 && csvs.length === 0 && (
+          <Alert theme="warning">
+            Charts not found. Manage your Helm chart repositories in Kubeapps by visiting the{" "}
+            <Link to={app.config.apprepositories(namespace)}>App repositories configuration</Link>{" "}
+            page.
+          </Alert>
```

Comment:
Why is this a warning? We can transform this into a "Welcome" message to start working with Kubeapps.

![A welcome message that suggest you the next step you need to accomplish to configure your application](https://user-images.githubusercontent.com/4056725/87532744-e1dbec00-c693-11ea-8991-081dbc801e1e.png)

Reply:
This is because if the catalog is empty, it's likely that something is going wrong. By default, kubeapps includes several repos so this should always have something but it's true that it can be the case in which a user deletes all repos.

Reply:
![Screenshot from 2020-07-15 18-21-53](https://user-images.githubusercontent.com/4025665/87569875-15386e00-c6c8-11ea-91b2-e003a9568955.png)